crates/ruff/src/rules/ruff/rules/pairwise_over_zipped.rs

```diff
@@ -89,12 +89,12 @@ fn match_slice_info(expr: &Expr) -> Option<SliceInfo> {
     ))
 }
 
-fn to_bound(expr: &Expr) -> Option<i64> {
+fn to_bound(expr: &Expr) -> Option<Integer> {
```

Comment:
A `i64` was probably sufficient to represent any reasonable value (any value outside of that range certainly results in an out of memory exception, assuming Python can even handle it)

Reply:
python is perfectly happy to do `list(range(10))[:2**10000]`. it's not reasonable from a user perspective but there's also no reason not to support it in ruff